ztools/x509/json.go

```diff
@@ -322,3 +337,18 @@ func (c *Certificate) MarshalJSON() ([]byte, error) {
 
 	return json.Marshal(jc)
 }
+
+func purgeNameDuplicates(names []string) (out []string){
```

Comment:
You can just use an `map[string]struct{}` for this (e.g. a map to nothing), and then iterating over it will give you a deduplicated list.